Added fix and unit test for queryContentRegexAssertion() method

RalfEggert opened the pull request:
Solves issue for #32.

Commit:
Added fix and unit test for queryContentRegexAssertion() method that did not check all matches

weierophinney commented on src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:
```diff
@@ -802,15 +802,22 @@ private function queryContentRegexAssertion($path, $pattern, $useXpath = false)
+            $nodeValues[] = $node->nodeValue;
+        }
+
+        if (!preg_match($pattern, implode('', $nodeValues))) {
```
The concern I have with this approach is that the regex might match across content boundaries.

As an example, consider this markup:

``` html
<div>foo</div>
<div>bar</div>
```

and then the following assertion:

``` php
$this->assertQueryContentRegex('div', '/foobar/');
```

The above, with the change you suggest here, would be a false positive.

I'd suggest doing the regex in the loop itself:

``` php
$found = false;

foreach ($result as $node) {
    if (preg_match($pattern, $node->nodeValue)) {
        $found = true;
        break;
    }
}

if (! $found) {
    throw new PHPUnit_Framework_ExpectationFailedException(/* ... */);
}

$this->assertTrue($found);
```

This approach will resolve the issue you reported in #32, while ensuring we
don't get false positives due to boundary issues.

weierophinney commented on test/PHPUnit/Controller/AbstractHttpControllerTestCaseTest.php:
```diff
@@ -414,6 +414,18 @@ public function testAssertQueryContentRegex()
         $this->assertQueryContentRegex('div#content', '#o{3,}#');
     }
 
+    public function testAssertQueryContentRegexMultipleMatches()
```
Following on the previous comment, please also add a test that validates that the markup I showed above would not pass this assertion.

Commit:
Added another test case to handle unexpected false positives for multiple matches

RalfEggert commented:
@weierophinney 

I added another test case to handle unexpected false positives and added for solution. I had to tweak it a little otherwise the exception message had to actual content to output. I hope that tweak is ok.

weierophinney commented:
Thanks, @RalfEggert! Cherry-picked to master, for release with 3.0.2.
